app/views/devise/registrations/_sign_up_async_request.js.erb

```diff
@@ -108,12 +108,19 @@ function validateEmailFormat(email) {
 	}
 }
 
-function validateEmailDomain(email){
-	var domain = email.split('@')
-	if(domain[1] != "schools.nyc.gov"){
+
+<% allowed_email_patterns = AllowedUserEmailMasks.where(active:true).pluck(:email_pattern) %>
+function validateEmailDomain(email) {
+	var domain = email.split('@');
```

Comment:
Do you want to get `.split('@')[-1]` then just `domain` below on line 118.

Reply:
The domain formula would return an empty string, if thing go wrong, and that would match everything.  So would need tweaking.